### ISSUE TITLE
big miss: dynamic time warping

### ISSUE THREAD

**Issue by m-r-munroe** [intel/scikit-learn-intelex#751]
**Describe the issue**
It is a miss in your benchmarking.  I think you should look at the "time-series-bakeoff" for others like it.
DTW is a pain, and very well known and well used method.  

The speedup you can get there is going to speak to signal analysts in several areas.

References for DTW:

- http://www.cs.ucr.edu/~eamonn/DTW_myths.pdf  <-- look at how DTW is described by Anthony Bagnall here
- https://arxiv.org/abs/1602.01711

Citation:    
Berndt, D. J., & Clifford, J. (1994, July). Using dynamic time warping to find patterns in time series. In KDD workshop (Vol. 10, No. 16, pp. 359-370).

**Comment by SmirnovEgorRu** [comment]
@m-r-munroe, it's interesting. What implementation of DTW do you (others) use? Some library?

**Comment by m-r-munroe** [comment]
pyts is decent.  
https://github.com/johannfaouzi/pyts

**Comment by m-r-munroe** [comment]
Along with random forests you should have done a grid-search using gradient boosted machines.  Those things take 10x more trees in the ensemble, and they are interdependent so they take a lot longer to run.  They used to be the winningest algorithm in kaggle.  

https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html
https://www.kaggle.com/msjgriffiths/r-what-algorithms-are-most-successful-on-kaggle/notebook 
https://bradleyboehmke.github.io/HOML/gbm.html

**Comment by syakov-intel** [comment]
@m-r-munroe could you please clarify the intent here? Are you looking for the benchmark or actual algorithm implementation?
